lib/renaissance/users/user.ex

```diff
@@ -2,33 +2,33 @@ defmodule Renaissance.User do
   use Ecto.Schema
   import Ecto.Changeset
 
+  @required_fields ~w(email password)a
+  @optional_fields ~w()a
+
   schema "users" do
     field :email, :string, unique: true
     field :password, :string, virtual: true
     field :password_hash, :string
+    has_many :auctions, Renaissance.Auction
 
     timestamps(type: :utc_datetime)
   end
 
   @doc false
   def changeset(user, attrs \\ %{}) do
     user
-    |> cast(attrs, [:email, :password])
-    |> validate_required([:email, :password])
+    |> cast(attrs, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
     |> unique_constraint(:email)
     |> hash_password()
   end
 
   defp hash_password(attrs) do
-    password = get_change(attrs, :password)
-
-    case password do
-      nil ->
-        attrs
-
-      _ ->
-        hash = Bcrypt.hash_pwd_salt(password)
-        attrs |> put_change(:password_hash, hash)
+    if is_nil(password = get_change(attrs, :password)) do
```

Comment:
The assignment in here is a little unconventional and unclear. I'm kind of surprised it's not a compile warning: https://github.com/elixir-lang/elixir/issues/7403

Reply:
I understood this to mean that it's unconventional & unclear because assignment occurs in the conditional, and so I addressed that here https://github.com/xray/Renaissance/commit/159a775e493da7136d042a4f76b690b4d8cb2aea#diff-3e2a5dd6d0ebb3b2d8b31401bf178663L28.

**If I misunderstood and did not address this, please let me know.**